src/integrations/admin/indexing-notification-integration.php

```diff
@@ -155,41 +155,28 @@ public function cleanup_notification() {
 	 * @return bool If the notification should be shown.
 	 */
 	protected function should_show_notification() {
-		if ( $this->is_indexation_warning_hidden() ) {
-			return false;
-		}
-
-		$indexation_reason         = $this->options_helper->get( 'indexables_indexation_reason', '' );
-		$indexation_started        = $this->options_helper->get( 'indexation_started', false );
-		$indexation_completed      = $this->options_helper->get( 'indexables_indexation_completed', false );
-		$ignore_indexation_warning = $this->options_helper->get( 'ignore_indexation_warning', false );
-
-		if ( $indexation_reason ) {
-			return true;
-		}
-
-		return (
-			$indexation_started === false &&
-			$indexation_completed === false &&
-			$ignore_indexation_warning === false
-		);
-	}
+		$indexing_reason = $this->options_helper->get( 'indexables_indexation_reason', '' );
 
-	/**
-	 * Returns if the indexation warning is temporarily hidden.
-	 *
-	 * @return bool True if hidden.
-	 */
-	protected function is_indexation_warning_hidden() {
-		if ( $this->options_helper->get( 'ignore_indexation_warning', false ) === true ) {
+		/*
+		 * Show a notification when we have a reason to do so.
+		 * For example when indexing has failed before and the user should try again.
+		 */
+		if ( $indexing_reason ) {
 			return true;
 		}
 
-		// When the indexation has started, but not completed.
+		/*
+		 * Do not show the notification when the indexation has started, but not completed.
+		 * I.e. when the user stopped it manually.
+		 */
 		if ( $this->options_helper->get( 'indexation_started', false ) > ( \time() - \MONTH_IN_SECONDS ) ) {
```

Comment:
`indexation_started` has a boolean value, so why are we comparing it against a point in time?

Reply:
Hmm, I see my above remark is incorrect, since `indexation_started` is assigned a time value in the `prepare` action. However, I can't seem to trigger this. Both when I let the indexation run until completion, or when I stop it in the middle, `indexation_started` remains at `0` for me in the database. When using the debugger, I don't seem to get past the `prepare` function in the code. In the network tab, I don't see a `prepare` request. Maybe we can have a look at this to make sure everything is working correctly.

Reply:
I copied this line form how it worked in the `Indexation_Integration`: https://github.com/Yoast/wordpress-seo/blob/trunk/src/integrations/admin/indexation-integration.php#L314. I agree that `false` is not great, though, so I changed it to `null`. In my opinion this more clearly indicates that its value has not been set yet.

I am looking into the bug you found.